network/openssl_privatekey.py

```
@@ -0,0 +1,225 @@
+        except IOError:
+            try:
+                os.remove(self.file_path)
+            except OSError:
+                pass
+            raise PrivateKeyError()
+
+    def remove(self):
+        '''Remove the private key from the filesystem.'''
+        if os.path.exists(self.file_path):
+            try:
+                os.remove(self.file_path)
+            except OSError:
+                raise PrivateKeyError()
+        else:
+            self.changed = False
+
+    def dump(self):
+        '''Serialize the object into a dictionnary.'''
+
+        result = {
+            'size': self.size,
+            'type': self.type,
+            'privatekey': self.file_path,
+            'changed': self.changed,
+        }
+
+        return result
+        
+
+def main():
+
+    module = AnsibleModule(
+        argument_spec = dict(
+            state=dict(default='present', choices=['present', 'absent'], type='str'),
+            name=dict(required=True, type='str'),
+            size=dict(default=4096, type='int'),
+            type=dict(default='RSA', choices=['RSA', 'DSA'], type='str'),
+            force=dict(default=False, type='bool'),
+            path=dict(required=True, type='path'),
+        )
+    )
+
+    if not pyopenssl_found:
+        module.fail_json(msg="the python pyOpenSSL module is required")
+
+    path = module.params['path']
+    if not os.path.isdir(path):
+        module.fail_json(name=path, msg='The directory %s does not exist or the file is not a directory' % path)
+
+    private_key = PrivateKey(module)
+    if private_key.state == 'present':
+        try:
+            private_key.generate()
+        except PrivateKeyError:
+            module.fail_json(name=path, msg='An error occured while generating the SSL private key')
+    else:
+        try:
+            private_key.remove()
+        except PrivateKeyError:
+            module.fail_json(name=path, msg='An error occured while removing the SSL private key')
```

It's not very friendly to hide the actual error (both type and message)!  How is the poor sysadmin supposed to figure out what went wrong (e.g. bad file permissions/out of disk space/pyopenssl parameter value out of supporter range/whatever else might happen)?